_includes/manuals/1.24/5.7.7_openid_sso.md

```diff
@@ -0,0 +1,106 @@
+
+OpenID provider provides an ID Token, encoded in JSON Web Token (JWT) to Foreman.
+
+An example of an encoded JWT ID Token:
+
+```
+eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c
+```
+
+#### Configuring Foreman
+
+As a system administrator, execute the following procedure to configure Foreman.
+
+Prerequisites
+
+* A working installation of Foreman at https://foreman.example.com.
+* A working installation of OpenID provider, for example, Keycloak at https://sso.example.com, which uses OIDC protocol. Ensure that the Keycloak server is running.
+* Ensure that both RP and OpenID provider are using https instead of http. If the certificates or the CA is self-signed, ensure that they are added to the End-user certificate trust store.
+
+Procedure
+
+1. Install the mod_auth packages:
+```
+# yum install mod_auth_openidc python2-requests-oauthlib keycloak-httpd-client-install -y
+```
+
+2. Register Katello/Foreman as application in the Keycloak:
+```
+# export KEYCLOAK_URL=https://keycloak.example.com
+# export KEYCLOAK_REALM=ssl-realm
+# export KEYCLOAK_USER=admin
+# keycloak-httpd-client-install --app-name foreman-openidc --keycloak-server-url $KEYCLOAK_URL --keycloak-admin-username $KEYCLOAK_USER --keycloak-realm $KEYCLOAK_REALM --keycloak-admin-realm master --keycloak-auth-role root-admin -t openidc -l /users/extlogin -d
+```
+Warning: This feature is not yet supported by foreman-installer. As a result, re-running the foreman-installer command can delete the changes in Apache files added by this feature.
+
+3. Log in to the Foreman admin page and click **Administer** → **Settings** → **Authentication**.
+ ![Foreman UI](Foreman_UI.png)
+
+4. Set **authorize_login_delegation_auth_source_user_autocreate** to **External** to enable auto-creation of users from external OpenID provider.
+
+5. Set the following OIDC parameters:
+
+ **OIDC Algorithm**: Algorithm type with which JWT was encoded in the OpenID provider. For example, For example, set this parameter to RS-256.
+
+ **OIDC Audience**: Name of the OIDC audience used for authentication. In case of [Keycloak](https://www.keycloak.org/), it is Client ID.
+
+ **OIDC Issuer**: The Issuer (iss) claim identifies the principal that issues the JWT.
+
+ **OIDC JWKS URL**: Open JSON Web key Set URL to validate the signature.
+
+ ![Foreman UI](Foreman_UI_Settings.png)
+
+#### End-user
+
+An end user can perform SSO by the following ways:
+* User Interface
+* Hammer-CLI
+
+To perform the mentioned ways, ensure that Foreman is configured. For more information, see Configuring Foreman.
+
+##### User Interface
+
+Procedure
+
+* Log in to the https://foreman.example.com/users/extlogin
+
+ If credentials are correct, it redirects to foreman dashboard.
```

Comment:
Foreman will automatically redirect to Keycloak Sign-in page. If credentials are correct, it redirects to Foreman dashboard.